docs/reference/ois/next/processing.md

```diff
@@ -77,14 +93,84 @@ post-processing.
 
 :::
 
+### v2
+
+The processing snippet receives parameters submitted as part of a template or
+on-chain request.
+
+Airnode executes snippets for `preProcessingSpecificationV2` and
+`postProcessingSpecificationV2` during its run cycle. The following describes
+the work flow Airnode uses:
+
+1. Run `preProcessingSpecificationV2`
+2. Airnode calls requested OIS endpoint
+3. Run `postProcessingSpecificationV2`
+4. Airnode encodes the response values defined by reservedParameters
+
+The processing schema is the same for both
+[`preProcessingSpecificationV2`](/reference/ois/next/specification.md#_5-11-preprocessingspecificationv2)
+and
+[`postProcessingSpecificationV2`](/reference/ois/next/specification.md#_5-12-postprocessingspecificationv2).
+Snippets for both specifications follow this schema:
+
+- `environment` - Currently only possible value is `Node`. This options
+  interprets the code as JavaScript and execute in Node.js. The function can be
+  also asynchronous (async/await is supported as well). The processing
+  implementation will wait for the function to resolve.
+- `value` - The processing code written as a string.
+- `timeoutMs` - The maximum timeout that this snippet can run. In case the
+  timeout is exceeded an error is thrown.
+
+Try the [Post processing](/guides/airnode/post-processing/) guide to further
+understand pre/post processing.
+
+#### Input and Output
+
+The processing snippet is a function which receives payload as an argument. The
+return value of the function is treated as a processing result. Apart from the
+payload argument, you can use most Node.js built-in modules.
+
+The payload argument for pre-processing is an object with the following
+properties:
+
+- `apiCallParameters` - The API call parameters with the exception of reserved
+  parameters. For example, if there was a parameter named `myParameter` defined
+  in the `endpoints[n].parameters` array, its value could be accessed using
+  `endpointParameters.myParameter` within pre-processing snippet.
+
+The output of the pre-processing snippet is an object with the following fields:
+
+- `apiCallParameters` - The pre-processed API call parameters. These are used to
```

Comment:
`operationParameters` should work here if we want to limit the vocabulary

Reply:
What I like though is that you receive `apiCallParameters` in the payload of the function and you can return the same thing. It's wierd when they have a different name.

Reply:
The semantic problem with mutating something is that you have two versions of the same thing that you call with the same name. For example, if the input is `endpointParameters`, calling the output `preprocessedEndpointParameters` would be more accurate than calling it `endpointParameters`. That being said, these endpoint parameters will be used as API operation parameters next, so they could be called that. `apiCallParameters` also works here, I'm just saying that since we're already talking about API operations a lot in the context of OIS (and in contrast "API call" is not introduced as a concept), it may be better to reuse that instead of adding a term.

Reply:
I dislike the `apiCallParameters` now and I slightly prefer `operationParameters` over `preprocessedEndpointParameters `.

Reply:
All options are kinda bad so I'm fine with both the input and the output being called `endpointParameters` here

Reply:
After reading on a bit, it didn't feel clear if `endpointParameters` in post-processing are raw endpoint parameters or pre-processed endpoint parameters, which is another downside of calling the output of this `endpointParameters`. As a counterargument to that, if we call the output of this `preprocessedEndpointParameters`, it will appear like endpoint parameters in OIS (https://docs.api3.org/reference/ois/latest/specification.html#_5-5-parameters) are mapped to raw endpoint parameters and not pre-processed endpoint parameters.
I guess simply mentioning in the post-processing docs that `endpointParameters` are raw/un-preprocessed is the best solution to this.

Reply:
Leaving them as `endpointParameters` and clarifying the `endpointParameters` in the post-processing section.